### ISSUE TITLE
DiscoveryJerseyProvider: Error creating a serializer.  ClassNotFoundException: jackson

### ISSUE THREAD

**Issue by spencergibb** [Netflix/eureka#702]
```
2015-11-06 12:50:15.910 ERROR 81323 --- [et_localhost-13] c.n.d.provider.DiscoveryJerseyProvider   : Error creating a serializer.

java.lang.ClassNotFoundException: jackson
	at java.net.URLClassLoader$1.run(URLClassLoader.java:372) ~[na:1.8.0_05]
	at java.net.URLClassLoader$1.run(URLClassLoader.java:361) ~[na:1.8.0_05]
	at java.security.AccessController.doPrivileged(Native Method) ~[na:1.8.0_05]
	at java.net.URLClassLoader.findClass(URLClassLoader.java:360) ~[na:1.8.0_05]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_05]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308) ~[na:1.8.0_05]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_05]
	at java.lang.Class.forName0(Native Method) ~[na:1.8.0_05]
	at java.lang.Class.forName(Class.java:259) ~[na:1.8.0_05]
	at com.netflix.discovery.provider.DiscoveryJerseyProvider.getSerializer(DiscoveryJerseyProvider.java:207) [eureka-client-1.3.4.jar:1.3.4]
	at com.netflix.discovery.provider.DiscoveryJerseyProvider.readFrom(DiscoveryJerseyProvider.java:118) [eureka-client-1.3.4.jar:1.3.4]
	at com.sun.jersey.api.client.ClientResponse.getEntity(ClientResponse.java:634) [jersey-client-1.19.jar:1.19]
	at com.sun.jersey.api.client.ClientResponse.getEntity(ClientResponse.java:586) [jersey-client-1.19.jar:1.19]
	at com.netflix.eureka.transport.JerseyReplicationClient.submitBatchUpdates(JerseyReplicationClient.java:120) [eureka-core-1.3.4.jar:1.3.4]
	at com.netflix.eureka.cluster.ReplicationTaskProcessor.process(ReplicationTaskProcessor.java:71) [eureka-core-1.3.4.jar:1.3.4]
	at com.netflix.eureka.util.batcher.TaskExecutors$BatchWorkerRunnable.run(TaskExecutors.java:187) [eureka-core-1.3.4.jar:1.3.4]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_05]
```

It fails with this class
```
@Serializer("jackson") // For backwards compatibility with DiscoveryJerseyProvider
public class ReplicationListResponse {
```

I'm using eureka 1.3.4

**Comment by spencergibb** [comment]
```
2015-11-06 12:00:15.733  INFO 81323 --- [ost-startStop-1] c.n.d.provider.DiscoveryJerseyProvider   : Using encoding codec LegacyJacksonJson
2015-11-06 12:00:15.734  INFO 81323 --- [ost-startStop-1] c.n.d.provider.DiscoveryJerseyProvider   : Using decoding codec LegacyJacksonJson
```

**Comment by tbak** [comment]
Looking into the source code, this code path is executed only when the
injected decoder is not capable of handling the media type. It happens in
the replication path which should always use JSON encoding. There is a bug
in the annotation value in ReplicationList/ReplicationResponse, which
contains "jackson", not the class name.

Can you check what encoding format is used on the wire?

/Tomasz

**Comment by spencergibb** [comment]
in `JerseyReplicationClient.submitBatchUpdates(JerseyReplicationClient.java:120` it sets 
```
                    .accept(MediaType.APPLICATION_JSON_TYPE)
                    .type(MediaType.APPLICATION_JSON_TYPE)
```

**Comment by tbak** [comment]
Which is strange the injected decoder does not handle this. We have this
code version running in the cloud with no such issue.
Can you check with the debugger why the ISerializer is being used
in DiscoveryJerseyProvider?

decoder.support(mediaType) returns false, so either Content-Type is not
defined in response, or it is not application/json.

**Comment by spencergibb** [comment]
The problem happens when the mediaType is `application/json; charset=UTF-8`. Then it drops to the `ISerializer`.

**Comment by tbak** [comment]
I will fix this, and make a new release, but probably not earlier than on
Monday.

**Comment by spencergibb** [comment]
I think I found in our code where the charset get's forcefully appended.

**Comment by tbak** [comment]
Lets keep it open, as this problem may occur again and the root cause is not obviously visible.
A simple fix would be to prevent any parameters in Content-Type header, and return a meaningful error reply.

**Comment by tbak** [comment]
Released in version 1.4.2
